progress/tracker.go

```diff
@@ -63,6 +64,14 @@ func (t *Tracker) IsDone() bool {
 	return t.done
 }
 
+// Error returns an error if MarkAsErrorDone() was called with an error
+func (t *Tracker) Error() error {
```

Comment:
Since we are not using the `err` object anywhere? Do you just wanna call this `IsErrored() bool` and change the signature of `MarkAsErrordone(err error)` to `MarkAsErrored()`?

Reply:
The intent was that a user program could use this to retrieve the error that the tracker failed with.

Reply:
Ok, but using the tracker to communicate the `error` object might be the wrong way to go about this for any client.

Reply:
I'm not convinced, but that's fine. Fixed.